### PR TITLE
Make some more file utils public

### PR DESCRIPTION
**Pull request by MarcoKoch**
See #1509.
I added those functions from `util.files` to `tools` that I think make sense from a conan user's point of view. There will be a separate pull request to the docs repo that adds documentation for them.

### CODE REVIEW

File: conans/tools.py [hunk 1]
```diff
@@ -22,7 +22,8 @@
 from conans.errors import ConanException
 from conans.model.version import Version
 # noinspection PyUnresolvedReferences
-from conans.util.files import _generic_algorithm_sum, load, save, sha256sum, sha1sum, md5sum, md5
+from conans.util.files import _generic_algorithm_sum, load, save, sha256sum, \
+    sha1sum, md5sum, md5, touch, relative_dirs, rmdir, mkdir, path_exists
```
Comment (memsharded) [inline conans/tools.py]: I am not sure, ``path_exists`` and ``relative_dirs`` might be too much for conan internals, do you think they provide value for package creators?

Comment (MarcoKoch) [inline conans/tools.py]: I think, `relative_dirs` might be handy for e.g. copying all license files out of a directory tree. Though I am with you with `path_exists`, now that I worked on the documentation. It seems a bit too specialized.

**Comment by memsharded** [comment]
Perfect! Will merge when CI is finished, it will be in 0.25, to be released this week.
Many thanks!

**Comment by MarcoKoch** [comment]
Nice! Thank you too.
The docs will be there too in a moment ;)

**Pull request merged**